cleverhans/attacks.py

```diff
@@ -33,7 +34,23 @@ def __init__(self, model, back='tf', sess=None):
         self.model = model
         self.back = back
         self.sess = sess
-        self.inf_loop = False
+
+        # We are going to keep track of old graphs and cache them.
+        self.graphs = {}
+
+        # When calling generate_np, arguments in the following set should be
+        # fed into the graph, as they are not structural items that require
+        # generating a new graph.
+        # This dict should map names of arguments to the types they should
+        # have.
+        # (Usually, the target class will be a feedable keyword argument.)
+        self.feedable_kwargs = {}
+
+        # When calling generate_np, arguments in the following set should NOT
+        # be fed into the graph, as they ARE structural items that require
+        # generating a new graph.
+        # This list should contain the names of the structural arguments.
+        self.structural_kwargs = []
```

Comment:
I'd make this an OrderedDict to reduce the amount of variability between different runs of the same script